From dfb80ffb176e61302912256f0c41496628f42d8f Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 13:06:24 -0400
Subject: [ethernet] qlcnic: Remove casts of pointer to same type

Message-id: <1408626422-3941-101-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88619
O-Subject: [PATCH RHEL 7.1 100/138] qlcnic: Remove casts of pointer to same type
Bugzilla: 1089352
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Joe Perches <joe@perches.com>

Bug 1089352 (https://bugzilla.redhat.com/show_bug.cgi?id=1089352)

net commit id 4e2e865d959e095ab8f1a112e7952c9baa173d0a

Casting a pointer to a pointer of the same type is pointless,
so remove these unnecessary casts.

Done via coccinelle script:

$ cat typecast_2.cocci
@@
type T;
T *foo;
@@

- (T *)foo
+ foo

Signed-off-by: Joe Perches <joe@perches.com>
Acked-by: Shahed Shaikh <shahed.shaikh@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_init.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_init.c
index ec399b7..2d91975 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_init.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_init.c
@@ -1388,7 +1388,7 @@ static int qlcnic_83xx_copy_fw_file(struct qlcnic_adapter *adapter)
  addr = (u64)dest;
 
  ret = qlcnic_83xx_ms_mem_write128(adapter, addr,
-       (u32 *)p_cache, size / 16);
+       p_cache, size / 16);
  if (ret) {
   dev_err(&adapter->pdev->dev, "MS memory write failed\n");
   release_firmware(fw);
-- 
1.7.1